### PR TITLE
Fix inconsistency in forms docs + add note about uncontrolled components

### PR DESCRIPTION
**Pull request by Zwyx**
Following discussion [here](https://github.com/shadcn/ui/issues/410).

Please feel free to ask me to change anything, or to make any change yourself in this PR.

Thanks for your work :slightly_smiling_face:

### CODE REVIEW

File: apps/www/content/docs/forms/react-hook-form.mdx [hunk 1]
```diff
@@ -103,7 +103,7 @@ import Link from "next/link"
 import * as z from "zod"
 
 const formSchema = z.object({
-  name: z.string().min(2).max(50),
+  username: z.string().min(2).max(50),
```
Comment (Zwyx) [inline apps/www/content/docs/forms/react-hook-form.mdx:106]: `username` is used in the rest of the form, not `name`.

File: apps/www/content/docs/forms/react-hook-form.mdx [hunk 2]
```diff
@@ -142,6 +142,8 @@ export function ProfileForm() {
 }
 ```
 
+Note that default values for each field are required. Without them, they will be [uncontrolled components](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable), and React will error.
```
Comment (Zwyx) [inline apps/www/content/docs/forms/react-hook-form.mdx:145]: This is to be consistent with the note at line 87. Feel free to let me know if you'd prefer not to receive PRs of small importance ;-)

Comment (shadcn) [reply apps/www/content/docs/forms/react-hook-form.mdx:145]: Can we rephrase this to:

Since FormField is using a controlled component, you need to provide a default value for  the field. See the [React Hook Form docs](https://react-hook-form.com/api/usecontroller) to learn more about controlled components.

Comment (Zwyx) [reply apps/www/content/docs/forms/react-hook-form.mdx]: Of course! Done ;-)
